src/storage_utils.cpp

```diff
@@ -543,11 +543,19 @@ std::int64_t get_filesize(stat_cache& stat, file_index_t const file_index
 			{
 				if (fs.pad_file_at(file_index)) continue;
 
+				// files with priority zero may not have been saved to disk at their
+				// expected location, but is likely to be in a partfile. Just exempt it
+				// from checking
+				if (file_index < file_priority.end_index()
+					&& file_priority[file_index] == dont_download
```

Comment:
I wonder what happens if user deselects some files (i.e. set download priority to 0) and chooses "skip hash checking" (i.e. enables "seed mode")? Can we consider this use case as valid? What behavior user expects to see?

Reply:
I just tried adding a torrent which contained 43 files.

1 movie file
1 .txt file
1 .exe file
40 .srt files

Deselected 38 .srt files/1 .exe file & 1 .txt file & checked `skip hash checking`

logs immediately showed `07/08/2020 12:24 - File sizes mismatch for torrent 'ABCD1234', pausing it.`

qBittorrent GUI status shows `missing files`

![skip hash check missing files](https://user-images.githubusercontent.com/42386382/89641811-6ba16280-d8aa-11ea-914a-578c35b1846a.png)

Reply:
@glassez In your scenario, does the user already have the selected files on their system or not?

If not, I would expect @xavier2k6's result. If they do, then I would not expect that; instead, I would expect the torrent to show up as completed without errors.

Reply:
I myself never use this feature, but I assume that all the user wants from it is to add some torrent for seeding using existing files, but wants to skip the hash check (due to big file size, for example). It may never download all the garbage files contained in many torrents, so it may not have all the torrent files (as in @xavier2k6 example), but it still want to add corresponding torrent for seeding only existing part of files and skip hash checking.

Reply:
@glassez

>  but I assume that all the user wants from it is to add some torrent for seeding using existing files, but wants to skip the hash check (due to big file size, for example). 

Yes, I also believe this is the main use case for this feature. I would expect it to work (i.e., going straight to "Seeding") both when all files are selected and when only some files are selected, as long as said files exist on the system of course.

I think in https://github.com/arvidn/libtorrent/pull/4948#discussion_r466989767 @xavier2k6 did not have any of the files (so immediately transitioning to the "missing files" state is the expected result). @xavier2k6 can you please confirm this was the case?

Reply:
>but I assume that all the user wants from it is to add some torrent for seeding using existing files, but wants to skip the hash check (due to big file size, for example).

>Yes, I also believe this is the main use case for this feature. I would expect it to work (i.e., going straight to "Seeding") both when all files are selected and when only some files are selected, as long as said files exist on the system of course.

That is what most users would like to do alright, but..........checking this now shows missing files when adding a previously completed torrent via deselection of unwanted files/enabling `skip hash check` in GUI status/`File sizes mismatch for torrent 'ABCD1234', pausing it`

HOWEVER! - there seems to be slight **"DATA LOSS"**

I used the same torrent for testing purposes as previously. (43 files)

The torrent is no longer @ 100% (It's 99.8% in transfer list window but in contents view "movie file" shows at 99.9%)

![complete torrent added](https://user-images.githubusercontent.com/42386382/89674438-08cabe00-d8e0-11ea-88ef-900d91eae422.png)


The previously selected .srt files x2 are 100% but the main "movie file" is showing now as having 84.6 KiB remaining.

![contents tab](https://user-images.githubusercontent.com/42386382/89674479-2009ab80-d8e0-11ea-9e96-7f95d535bf32.png)

Pieces missing?

![Test Info](https://user-images.githubusercontent.com/42386382/89674636-62cb8380-d8e0-11ea-8c0c-41d5de3e648a.png)



@FranciscoPombal 
>I think in #4948 (comment) @xavier2k6 did not have any of the files (so immediately transitioning to the "missing files" state is the expected result). @xavier2k6 can you please confirm this was the case?

Yes, this was the case - confirm.